OpenEdXMobile/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -514,4 +528,90 @@ private static TextView createField(@NonNull LayoutInflater inflater, @NonNull V
+                            PackageManager.PERMISSION_GRANTED) {
+                        return false;
+                    }
+                    break;
+                case READ_STORAGE_PERMISSION_REQUEST:
+                    if (ContextCompat.checkSelfPermission(getActivity(), Manifest.permission.READ_EXTERNAL_STORAGE) !=
+                            PackageManager.PERMISSION_GRANTED) {
+                        return false;
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+        return true;
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    public void requestPermissions(int requestCode) {
+        switch (requestCode) {
+            case CAMERA_PERMISSION_REQUEST:
+                requestPermissions(new String[]{Manifest.permission.CAMERA}, CAMERA_PERMISSION_REQUEST);
+                break;
+            case READ_STORAGE_PERMISSION_REQUEST:
+                requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE}, READ_STORAGE_PERMISSION_REQUEST);
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+            case CAMERA_PERMISSION_REQUEST:
+                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    permissionGranted(CAMERA_PERMISSION_REQUEST);
+                } else {
+                    showSnackMessage(getResources().getString(R.string.permission_not_granted));
+                }
+                break;
+            case READ_STORAGE_PERMISSION_REQUEST:
+                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    permissionGranted(READ_STORAGE_PERMISSION_REQUEST);
+                } else {
+                    showSnackMessage(getResources().getString(R.string.permission_not_granted));
+                }
+                break;
+
+            default:
+                break;
+        }
+
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+    }
+
+    public void permissionGranted(int requestCode) {
+        switch (requestCode) {
+            case CAMERA_PERMISSION_REQUEST:
+                startActivityForResult(helper.createCaptureIntent(getActivity()), CAPTURE_PHOTO_REQUEST);
+                break;
+
+            case READ_STORAGE_PERMISSION_REQUEST:
+                final Intent galleryIntent = new Intent()
+                        .setType("image/*")
+                        .setAction(Intent.ACTION_GET_CONTENT);
+                startActivityForResult(galleryIntent, CHOOSE_PHOTO_REQUEST);
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    public void showSnackMessage(String message) {
```

Comment:
No need for this function. `Snackbar`'s make should be used itself at each occurrence.